### PR TITLE
Remove extensions support from the operator CRD

### LINKED ISSUE #10835: Remove built-in support for extensions in the operator
The operator is designed to provide production grade deployments of Keycloak. It is a best practice that production grade deployments use a custom image if extensions are required.
Reasoning:
 - Keycloak will not start if the provider of the Jar (e.g. GitHub) is down or not reachable
 - Downloading something on the fly on start breaks a bunch of patterns regarding immutability and idempotency
 - Slow down Keycloak start with little benefit

### PR DESCRIPTION
**Pull request by andreaTP**
Resolves #10835

### CODE REVIEW

File: operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java [hunk 1]
```diff
@@ -101,60 +99,6 @@ private Deployment fetchExistingDeployment() {
...
-        volumes.add(extensionVolume);
-        baseDeployment.getSpec().getTemplate().getSpec().setVolumes(volumes);
-
-        // Add the main deployment Volume Mount
-        var container = baseDeployment.getSpec().getTemplate().getSpec().getContainers().get(0);
-        var containerVolumeMounts = container.getVolumeMounts();
-
-        var extensionVM = new VolumeMountBuilder()
-                .withName(Constants.EXTENSIONS_VOLUME_NAME)
-                .withMountPath(Constants.KEYCLOAK_PROVIDERS_FOLDER)
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java:132]: Please remove the unused constants as well:
https://github.com/keycloak/keycloak/blob/1f10a4feb2dfea88bb62e76e314c4e999ef75cc9/operator/src/main/java/org/keycloak/operator/Constants.java#L41-L46

File: operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java [hunk 2]
```diff
@@ -101,60 +99,6 @@ private Deployment fetchExistingDeployment() {
...
-        containerVolumeMounts.add(extensionVM);
-
-        container.setVolumeMounts(containerVolumeMounts);
-
-        // Add the Extensions downloader init container
-        var extensionsValue = extensions.stream().collect(Collectors.joining(","));
-        var initContainer = new ContainerBuilder()
-                .withName(Constants.INIT_CONTAINER_NAME)
-                .withImage(config.keycloak().initContainerImage())
-                .withImagePullPolicy(config.keycloak().initContainerImagePullPolicy())
```
Comment (vmuzikar) [inline operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java:144]: Please remove also the unused config options:
https://github.com/keycloak/keycloak/blob/1f10a4feb2dfea88bb62e76e314c4e999ef75cc9/operator/src/main/resources/application.properties#L11-L12

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java (hunk 1 above)
Comment (andreaTP) [reply operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java:145]: 👍  done

File: operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java (hunk 2 above)
Comment (andreaTP) [reply operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java:157]: 👍  done

**Review by vmuzikar — CHANGES_REQUESTED** [review]
Ah, I forgot that we also need to remove this:
https://github.com/keycloak/keycloak/blob/6bce8b80b99ff9a0d3a13212d1fdb813f4c84390/operator/src/main/java/org/keycloak/operator/Config.java#L33-L34

**Comment by andreaTP** [comment]
@vmuzikar 
> Ah, I forgot that we also need to remove this:

just done 👍

**Review by vmuzikar — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**